openstf/model/serializer.py

```diff
@@ -0,0 +1,204 @@
+            trained_models_folder = custom_folder
+        else:
+            trained_models_folder = self.trained_models_folder
+
+        # Combine into complete folder path
+        model_folder = trained_models_folder / model_folder_name
+
+        return model_folder
+
+    def _find_most_recent_model_folder(self, pid_model_folder: Path) -> Path:
+        """Find the model recent model folder.
+            Iterate over the directories in the 'pid' model folder (the top level model
+            folder for a specific pid) and find the
+        Args:
+            pid_model_folder (pathlib.Path): [description]
+        Raises:
+            FileNotFoundError: When no model file is found in any of the subdirectories
+                of `pid_model_folder` which start with a 2.
+        Returns:
+            pathlib.Path: Path to the most recent model file
+        """
+
+        # Declare empty list to append folder names
+        model_folders = []
+
+        # Loop over all subfolder of pid_model_folder and store the ones starting with
+        # a 2 (date starts with 2000s)
+        for folder in pid_model_folder.iterdir():
+            if folder.name.startswith("2") is True and folder.is_dir() is True:
+                model_folders.append(folder)
+
+        # sort folders by date such that the most recent date is the first element
+        for folder in sorted(model_folders, reverse=True):
+            model_file = folder / MODEL_FILENAME
+            if model_file.is_file() is True:
+                model_folder = folder
+                break
+        # we did not find a valid model file
+        else:
+            msg = f"No model file found at save location: '{pid_model_folder}'"
+            self.logger.error(msg)
+            raise FileNotFoundError(msg)
+
+        return model_folder
+
+    def _build_save_folder_path(self) -> Path:
+        """Builds a save path where to save a model.
+
+        Returns: pathlib.Path() with location where to save a model
+
+        """
+
+        # Get specific folder for this precion job
+        pid_model_folder = self._build_pid_model_folder_path()
+
+        # Get the datetime string
+        datetime_str = datetime.utcnow().strftime(FOLDER_DATETIME_FORMAT)
+
+        # Combine into a complete model save path
+        save_folder = pid_model_folder / datetime_str
+
+        return save_folder
+
+    @staticmethod
+    def _determine_model_age_from_path(model_location: Path) -> float:
+        """Determines how many days ago a model is trained base on the folder name.
+
+        Args:
+            model_location: pathlib.Path: Path to the most recent model file
+
+        Returns: Number of days since training of the model
+
+        """
+
+        # If we cannot determine the age return infinite days
+        if model_location is None:
+            return float("inf")
+
+        # trained_models/<pid>/<YYYYMMDDHHMMSS>/
+        datetime_string = model_location.name
```

Comment:
Not sure if I get this. the `model_location` argument is the path to the most recent model file, that would be `trained_models/<pid>/<YYYYMMDDHHMMSS>/model.sav`. I don't see how that can be a datetime string.